packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -318,112 +318,133 @@ async function createViewUnparentedVideo(
   };
 }
 
-function disposeViewRemoteVideo(
+function disposeViewVideo(
   context: CallContext,
   internalContext: InternalCallContext,
   callId: string,
-  participantId: CommunicationIdentifierKind | string,
-  stream: RemoteVideoStreamState
+  stream: RemoteVideoStreamState | LocalVideoStreamState,
+  participantId?: CommunicationIdentifierKind | string
 ): void {
-  const streamId = stream.id;
-  const streamType = stream.mediaStreamType;
-  let participantKey;
-  if (typeof participantId === 'string') {
-    participantKey = participantId;
-  } else {
-    participantKey = toFlatCommunicationIdentifier(participantId);
-  }
-
-  const streamLogInfo = { callId, participantKey, streamId, streamType };
-
-  _logDisposeStreamEvent(EventNames.START_DISPOSE_REMOTE_STREAM, streamLogInfo);
+  // we can only have 3 types of createView
+  let disposeViewType: 'local' | 'remote' | 'unparented';
```

Comment:
same comments as other PR, for variables make them const and assign at declaration, and for logging functions combine them to avoid having the check for every log line